src/Components/ThemesMenu/ThemesMenu.jsx

```diff
@@ -0,0 +1,123 @@
+import { useState } from "react";
+import Button from "../Button/Button";
+import "./ThemesMenu.css";
+
+export function ThemesMenu({
+  currentTheme,
+  onChange,
+  themes,
+  onAddTheme,
+  onEditTheme,
+  onDeleteTheme,
+}) {
+  const [newThemeName, setNewThemeName] = useState("");
+  const [isEditingTheme, setIsEditingTheme] = useState(false);
+  const [editThemeName, setEditThemeName] = useState(currentTheme.name);
+  const [isDelitingTheme, setIsDeletingTheme] = useState(false);
+
+  const handleThemeNameChange = (event) => {
+    setNewThemeName(event.target.value);
+  };
+
+  const handleEditThemeNameChange = (event) => {
+    setEditThemeName(event.target.value);
+  };
+
+  const handleEditClick = () => {
+    setIsEditingTheme(true);
+    setEditThemeName(currentTheme.name);
+  };
+
+  const handleUpdateClick = () => {
+    onEditTheme({ ...currentTheme, name: editThemeName });
+    setIsEditingTheme(false);
+  };
+
+  const handleCancelClick = () => {
+    setIsEditingTheme(false);
+    setIsDeletingTheme(false);
+  };
+
+  const handleDeleteClick = () => {
+    setIsDeletingTheme(true);
+  };
+
+  const handleConfirmingDeleteClick = () => {
+    onDeleteTheme(currentTheme.id);
+    setIsDeletingTheme(false);
+  };
+
+  return (
```

Comment:
just make sure to add ` const isDefaultTheme = currentTheme.id === "t1";` before return , to check if the the selected theme is the defaultTheme ,then the user can't click on Edit and Delete buttons... you just need maybe to change the css of those two button....